### PR TITLE
Migrate to 3.4 #7259

### PR DESCRIPTION
**Pull request by priyankarani**

### CODE REVIEW

File: tests/test_templates.py [hunk 1]
```diff
@@ -1036,6 +948,11 @@ def test_0075_products(self):
             with app.test_client() as c:
                 self.create_test_products()
 
+                template1, = self.ProductTemplate.search([
+                    ('name', '=', 'product 1')
+                ])
+                print "????", template1.products[0].get_images()
```
Comment (tarunbhardwaj) [inline tests/test_templates.py]: ????

File: party.py [hunk 2]
```diff
@@ -26,7 +26,9 @@
 except ImportError:
     logging.error("pygeoip is not installed")
 else:
-    path = os.environ.get('GEOIP_DATA_PATH', CONFIG.get('geoip_data_path'))
+    path = os.environ.get(
```
Comment (prakashpp) [inline party.py]: make a section for `nereid_wesbhop`

File: tests/test_base.py [hunk 3]
```diff
@@ -124,9 +134,15 @@ def _create_product_template(
                 ('create', [{
                     'uri': uri,
                     'displayed_on_eshop': displayed_on_eshop,
+                    'code': code,
                 }])
             ]
-        return self.ProductTemplate.create(vlist)
+        template = self.ProductTemplate.create(vlist)[0]
+        self.file.product = template.products[0]
```
Comment (prakashpp) [inline tests/test_base.py]: this has been replaced with the `product.media`, check nereid-catalog latest code

**Comment by prakashpp** [comment]
Also task id is wrong

File: tests/test_gift_card.py [hunk 4]
```diff
@@ -417,7 +422,7 @@ def test0080_add_gift_card_to_cart_case2(self):
                 )
 
                 # Test login handler
-                self.login(c, 'email@example.com', 'password')
+                rv = self.login(c, 'email@example.com', 'password')
```
Comment (prakashpp) [inline tests/test_gift_card.py]: stray change

Comment (prakashpp) [inline tests/test_gift_card.py]: revert this change

**Comment by prakashpp** [comment]
@priyankarani thanks for the patch
